Src/Library/Extensions/ExpressionHelper.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Linq.Expressions;
+
+namespace FastEndpoints;
+
+
+internal static class ExpressionHelper
+{
+    internal static string GetPropertyChain(Expression expression)
+    {
+        return expression switch
+        {
+            MemberExpression m => BuildMemberChain(m),
+            BinaryExpression { NodeType: ExpressionType.ArrayIndex } be => FormatIndexerExpression(be.Left, be.Right),
+            MethodCallExpression { Object: not null } mce when IsItemAccessor(mce) => FormatIndexerExpression(mce.Object, mce.Arguments[0]),
+            UnaryExpression ue => GetPropertyChain(ue.Operand),
+            _ => throw new NotSupportedException($"[{expression}] is not a supported expression type!")
+        };
+    }
+
+    private static bool IsItemAccessor(MethodCallExpression mce)
+        => mce.Method.Name == "get_Item" && mce.Arguments.Count == 1;
+
+    private static string FormatIndexerExpression(Expression objectExpression, Expression indexExpression)
+    {
+        if (objectExpression is null)
+            throw new ArgumentNullException(nameof(objectExpression), "Object expression cannot be null.");
+
+        return $"{GetPropertyChain(objectExpression)}[{EvaluateExpression(indexExpression)}]";
+    }
+
+    private static string EvaluateExpression(Expression expression)
+        => Expression.Lambda(expression).Compile().DynamicInvoke() switch
```

Comment:
is there really no alternative to recursively compiling the expression on each call? i suppose this would be the biggest perf hit. 

the other two alternatives you linked seems to be from the MVC view stuff, which would have to be added as dependencies to the project. so i don't think we can go that route.

is it not possible to maybe cache the compiled expressions and re-use on subsequent calls?

Reply:
maybe as a compromise we just output `"ObjectArray[x].Test"` when the index is not a constant?

Reply:
Hmm 🤔 I can do some research and benchmarking of different solutions

Only displaying constant args can be a good compromise